### DISCUSSION TITLE
Type for handler with a parameter isn't inferred correctly

### DISCUSSION THREAD

**Discussion by TheOnlyTails** [honojs/hono#2097] [Q&A]
I have a simple handler like so:
```ts
app
	.get("/fonts", async (c) => {
		return c.json(await fetchWithZod(versions.v1.schemas.fonts, rawFile("v1", "fonts.json")));
	})

	.get("/fonts/:font", zValidator("param", z.object({ font: z.string() })), async (c) => {
		const data = await fetchWithZod(versions.v1.schemas.fonts, rawFile("v1", "fonts.json"));
		const sign = data[c.req.param("font")];

		return sign ? c.json(sign) : c.notFound();
	});
```
However, when I try to create a client from it, the return type from the `/fonts` handler is correct, but the `/fonts/:font` is resolved just to `{}`:
![image](https://github.com/honojs/hono/assets/65342367/1afa1788-0907-48cc-b3e9-6a179d21cdcf)

Why is this happening?

**Comment by EdamAme-x** [comment]
hi @TheOnlyTails 
`sign ? c.json(sign) : c.notFound();`
There may be a problem here.

**Reply by TheOnlyTails** [reply]
what's the problem with it?

**Reply by EdamAme-x** [reply]
I am not checking the detailed code,
I think not guessing correctly, since the client may return a response with a 404 status code or a normal response.

And there is another concern.
I would like to know the results of registering handlers in the opposite order.

**Reply by TheOnlyTails** [reply]
Registering the parameterized route first doesn't seem to change anything.

**Reply by TheOnlyTails** [reply]
I tried throwing an `HTTPException` instead of returning `notFound` when the param doesn't exist in the fetched data object, and now, for some reason, creating the client in dev works fine, but once I build it and look at the [.d.ts output](https://cdn.jsdelivr.net/npm/@kulupu-linku/sona@0.1.3/dist/client.d.ts), it now has `never` for _all_ the routes, not just the parameterized ones.

**Reply by yusukebe** [reply]
Use `c.json()` since `c.notFound()` does not support the RRC-mode.
